docs/Why-Mock.md

```diff
@@ -0,0 +1,107 @@
+# Why Mock?
+
+This document tries to answer two questions:
+
+1. Why you should use Mock instead of lower-level tooling like `rpmbuild`
+2. Why you should not try to reimplement Mock and rather join the project
+
+
+## Mock builds packages inside an isolated environment
```

Comment:
I'd prefer calling it "separated" rather than "isolated" because Mock isn't in charge of isolation.

Reply:
It's not separate either, because we don't use VMs to create separation (we have had plenty of bugs related to leakage from the host). So it's probably better to just say "dedicated environment".